README.md

```diff
@@ -406,16 +406,26 @@ where:
 
 By default, suggestions are rendered when the input isn't blank. Feel free to override this behaviour.
 
-This function gets the current value of the input, and it should return a boolean.
+This function gets the current value of the input and the reason why the suggestions might be rendered, and it should return a boolean.
 
 For example, to display suggestions only when input value is at least 3 characters long, do:
 
 ```js
-function shouldRenderSuggestions(value) {
+function shouldRenderSuggestions(value, reason) {
   return value.trim().length > 2;
 }
 ```
 
+You can use the second `reason` argument to finely control exactly when the suggestions are rendered. The possible values are closely related to those for `onSuggestionsFetchRequested`, plus a few extra cases:
```

Comment:
Do you think a "Reason" enum is overkill? It probably is... But it could be good so we make sure these reasons are in-sync.

Reply:
It's personal taste I guess, but personally I'd prefer keeping them as string values rather than formally making them an enum. I could easily extract each string reason into a constant though, which would help us enforce consistency, what do you think?

We should put fixed list of reason strings into the TypeScript either way too. That'd help for TS obviously, but should also get picked up by pure JS usage in VS code and similar I think, so these will be suggested appropriately. I'll make a PR over there once this one is merged.

Reply:
> I could easily extract each string reason into a constant though, which would help us enforce consistency, what do you think?

Sounds good, if it's not too much work for you. 

> We should put fixed list of reason strings into the TypeScript either way too. 

Where do we have TypeScript? I though all of this git was plain JS.

Reply:
> Sounds good, if it's not too much work for you.

Not at all, done :+1:

> Where do we have TypeScript? I though all of this git was plain JS.

Oh yes, this repo is, but there's types available as `@types/react-autosuggest` on npm, from here:

https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react-autosuggest

All TS users will be using those, and VS Code (among others) will [automatically](https://code.visualstudio.com/docs/nodejs/working-with-javascript#_typings-and-automatic-type-acquisition) find and use those for autocomplete and warnings even in JS.

Reply:
Thanks for the pointer. Interesting. (I'm just using Sublime, without type checking, so far...)